### PR TITLE
Interface Documentation

### LINKED ISSUE #4: Document the interface
I was going to build another executor (for Jenkins), but I don't know what I'm supposed to implement.

Please document:
 - Interface
 - Expected Execution (do I run a docker container, which one, what about secrets?)

### PR DESCRIPTION
**Pull request by nkatzman**
This PR adds in documentation for describing the start/stream interface

Describes:
1. Required parameters
2. Expected outcome
3. Expected callback

Resolves #4

### CODE REVIEW

**Comment by jer** [comment]
Could we also give a brief synopsis of what the executor is for?

**Comment by nkatzman** [comment]
@jer added

**Comment by jer** [comment]
👍

**Comment by jer** [comment]
lookin good 
2cxwcqs

**Comment by shruthi-venkateswaran** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -3,12 +3,60 @@
...
 > Base class defining the interface for executor implementations
 
+An executor is an engine that is capable of running a set of docker containers together.
+
+i.e. Jenkins, Kubernetes, ECS, Mesos
+
+The intention of an executor is to run the `launch` script defined in the [screwdriver job-tools] docker container, which is mounted to a container defined by a screwdriver [task]
+
+This means:
+1. Mounting the job-tools container as a volume to $MOUNT_POINT on another container
```
Comment (tkyi) [inline README.md]: Needs more spacing around 1. and 2. for proper formatting I think.

File: README.md [hunk 2]
```diff
@@ -3,12 +3,60 @@
...
+| config.container | String | Container for the build to run in |
+| config.scmUrl | String | The scmUrl to checkout |
+| callback | Function | Callback for when task has been created |
+
+##### Expected Outcome
+The start function is expected to create a [task] in the execution engine.
+
+##### Expected Callback
+1. When an error occurs, `callback(err)`
+2. When tasks is created correctly, `callback(null)`
```
Comment (tkyi) [inline README.md]: `When the task is created correctly,`

File: README.md [hunk 3]
```diff
@@ -3,12 +3,60 @@
...
+| config           | Object | Configuration Object |
+| config.buildId   | String | The unique ID for a build |
+| callback         | Function | Callback for when stream has been created |
+
+##### Expected Outcome
+The stream function is expected to return a readable stream upon success
+
+##### Expected Callback
+1. When an error occurs, `callback(err)`
+2. when stream is created correctly, `callback(null, stream)`
```
Comment (tkyi) [inline README.md]: `When the stream is created correctly`

**Comment by tkyi** [comment]
👍  LGTM

**Pull request merged**